### ISSUE TITLE
Illeagal instruction (core dumped)

### ISSUE THREAD

**Issue by sjparkmk** [bluenote-1577/fairy#1]
hi there, 
based on the published paper, i installed your fairy via conda. my system spec is ubuntu 20.04 and conda version 24.3.0
however, when typed single word "fairy", the program is showing the message "Illegal instruction (core dumped)"
(how) can i fix it?
cheers,

**Comment by bluenote-1577** [comment]
@sjparkmk hmm, I'm pretty sure this is related to a recent comment similar to another software of mine: https://github.com/bluenote-1577/sylph/issues/2#issuecomment-2082743443. The issue is that bioconda had some build issues... 

I will release a new bioconda version and hopefully it should be okay. 

For now, the source compile and static binary option should work.

Edit: a new version was released on bioconda but it's still broken. Not sure what to do... I'll investigate more when I have time

**Comment by sjparkmk** [comment]
thanks for your reply. option 3 is clearly working with fairy 0.5.4 !!

**Comment by bluenote-1577** [comment]
Future readers: I think I found the issue. 

**Right now,  AVX512 is needed for conda install**

The issue: fairy uses native CPU instructions for compilation. Conda compiles fairy on AVX512 machines -> autovectorization -> your machine can't run fairy :(

I will probably change this behaviour at some point... but the other 2 options will work no problem.

**Comment by LottePronk** [comment]
Hi, 
I'm facing the same issue. I would like to integrate fairy into a pipeline, but that would require a conda package that works with any CPU. Is there (still) a plan to make a conda package that works with other CPU instructions?

**Comment by bluenote-1577** [comment]
@LottePronk Yes I will update to v0.5.7 soon. This should work with conda now... hopefully.

**Comment by bluenote-1577** [comment]
@LottePronk I updated the fairy install on bioconda. can you let me know if there are issues with older CPUs, if you have them available? 

Thanks,
Jim

**Comment by LottePronk** [comment]
@bluenote-1577 I only tried it on one CPU, but it is working now. Thank you very much for fixing this so quickly! 

Cheers, 
Lotte
